### PR TITLE
Change molotov use action to transform

### PR DESCRIPTION
**Pull request by Gstayton**
Remove iuse::molotov, as it is no longer used

Bring molotovs in line with how pipebombs work, as a first step towards
normalizing various grenade-style items

This allows for more flexibility with the player messages on interacting with molotovs as well, given the original molotov function lacked these features.

### CODE REVIEW

File: data/json/items/tools.json [hunk 1]
```diff
@@ -3414,7 +3414,15 @@
     "ammo": "NULL",
     "revert_to": "null",
     "category": "weapons",
-    "use_action": "MOLOTOV"
+    "use_action": {
+        "type": "transform",
+        "target": "MOLOTOV_LIT",
```
Comment (illi-kun) [inline data/json/items/tools.json]: I think it's better to write it in the same manner as target item id (in lowercase : ```molotov_lit```).

**Comment by illi-kun** [comment]
Very nice change!

File: data/json/items/tools.json [hunk 2]
```diff
@@ -3414,7 +3414,15 @@
...
+        "target_charges": 1,
+        "active": true,
+        "need_fire": 1,
+        "need_fire_msg": "You need a source of fire!",
+        "msg": "You light the molotov cocktail"
```
Comment (illi-kun) [inline data/json/items/tools.json]: Use the message from replaced function: ```You light the Molotov cocktail.``` (your message missed uppercase 'M' and period at the end of the sentence).

File: src/iuse.cpp [hunk 3]
```diff
@@ -4954,24 +4954,6 @@ int iuse::arrow_flamable(player *p, item *it, bool, const tripoint& )
     return 1;
 }
 
-int iuse::molotov(player *p, item *it, bool, const tripoint& )
-{
-    if (p->is_underwater()) {
```
Comment (illi-kun) [inline src/iuse.cpp]: Can we light it underwater now? :-)

File: data/json/items/tools.json (hunk 1 above)
Comment (Gstayton) [inline data/json/items/tools.json]: You're right. I had changed it back, just not in my actual workspace.... Thanks for pointing that out.

File: src/iuse.cpp (hunk 3 above)
Comment (Gstayton) [inline src/iuse.cpp]: That was something I was worried about, but currently nothing using iuse::transform is capable of being used underwater; doesn't matter if it uses fire or not. I was debating opening an issue on that, but we really don't have enough underwater content.

Comment (illi-kun) [inline src/iuse.cpp]: Ah, I see, code in ```iuse_actor.cpp:L69``` prevents any transform action underwater. Good to know that.

File: data/json/items/tools.json [hunk 4]
```diff
@@ -3414,7 +3414,15 @@
...
     "category": "weapons",
-    "use_action": "MOLOTOV"
+    "use_action": {
+        "type": "transform",
+        "target": "molotov_lit",
+        "target_charges": 1,
+        "active": true,
+        "need_fire": 1,
+        "need_fire_msg": "You need a source of fire!",
+        "msg": "You light the Molotov cocktail"
```
Comment (illi-kun) [inline data/json/items/tools.json]: You still missed period (".") at the end.

**Comment by illi-kun** [comment]
I think you should remove line 129 from ```iuse.h``` as well:
```int molotov             ( player*, item*, bool, const tripoint& );```

**Comment by illi-kun** [comment]
Your changes work fine but during the test I found one weird thing (which is not your fault):
It's possible to activate lit Molotov cocktail more and more, it increases number of charges but I see no reason for doing that.

```
int iuse::molotov_lit(player *p, item *it, bool t, const tripoint &pos)
 {
     int age = int(calendar::turn) - it->bday;
     if (p->has_item(it)) {
         it->charges += 1; // <--- this code is responsible for this weird behavior
         if (age >= 5) { // More than 5 turns old = chance of going out
             if (rng(1, 50) < age) {
                 p->add_msg_if_player(_("Your lit Molotov goes out."));
                 it->make("molotov");
                 it->active = false;
             }
         }
     } else {
<...>
         }
     }
     return 0;
 }
```

File: data/json/items/tools.json [hunk 5]
```diff
@@ -3414,7 +3414,15 @@
...
-    "use_action": "MOLOTOV"
+    "use_action": {
+        "type": "transform",
+        "target": "molotov_lit",
+        "target_charges": 1,
+        "active": true,
+        "need_fire": 1,
+        "need_fire_msg": "You need a source of fire!",
+        "msg": "You light the Molotov cocktail."
+    }
```
Comment (Coolthulhu) [inline data/json/items/tools.json]: Old behavior had a move cost associated with it.
150 was a fair bit too much (currently lighting a rag on the ground takes like 25 or so), but is that removal intended?

Comment (Gstayton) [inline data/json/items/tools.json]: Was actually not intended. Suppose I shouldn't be making PR's at my equivalent of 3am on an empty stomach.
